### ISSUE TITLE
* identifier in argument list causes false positive for missing-type-doc.

### ISSUE THREAD

**Issue by Xyern** [PyCQA/pylint#3092]
When an * identifier is used in the argument list to require that all following keyword args need to be called explicitly, pylint gives false positives for the remaining keyword args via the missing-type-doc checker.

### **Steps to reproduce**
Execution on Mac OSX 10.14.6
Create a file pylint_failure.py. with the following:

```
def indentifier_kwarg_method(arg1: int, arg2: int, *, value1: str, value2: str):
    """Code to show failure in missing-type-doc

    Args:
        arg1: First argument.
        arg2: Second argument.
        value1: First kwarg.
        value2: Second kwarg.
    """
    print("NOTE: It doesn't like anything after the '*'.")
```
Test the above code for missing-type-doc.
(NOTE: I didn't test this with Python's latest addition of the / in the argument list, which may have a similar issue?)

### **Current Behaviour**
```
python3 -m pylint --disable=all --enable=missing-type-doc pylint_failure.py 
************* Module pylint_failure
pylint_failure.py:1:0: W9016: "value1, value2" missing in parameter type documentation (missing-type-doc)
```

### **Expected Behaviour**
The documented arguments should not be getting flagged as undocumented.

### **python3 -m pylint --version**
__main__.py 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Jun 19 2019, 07:38:49) 
[Clang 10.0.1 (clang-1001.0.46.4)]

**Comment by PCManticore** [comment]
Thanks @Xyern for opening an issue, I can reproduce it as well.
